base/compiler/optimize.jl

```diff
@@ -311,6 +311,13 @@ function statement_cost(ex::Expr, line::Int, src::CodeInfo, sptypes::Vector{Any}
                 ftyp = argextype(farg, src, sptypes, slottypes)
             end
         end
+        # Give calls to YAKCs zero cost. The plan is for these to be a single
+        # indirect call so have very little cost. On the other hand, there
+        # is enormous benefit to inlining these into a function where we can
+        # see the definition of the YAKC. Perhaps this should even be negative
```

Comment:
Do you want to encourage inlining a yakc into its caller, or the function containing the call into *its* caller? (perhaps both?)

Reply:
Both, e.g. consider:
```
call(f::YAKC) = f()
bar() = call(_yakc(...))
```
We would like to elide the allocation of the yakc and just inline the whatever the code the yakc would have had right into the caller, but in order to be able to do that, we need to see both the call and the definition to the yakc, so a high inline penalty for calling a yakc would be counterproductive.